tools/tests/test_cfs_card.py

```
@@ -0,0 +1,366 @@
+    """Guards against using display strings instead of raw states."""
+    sensor = (
+        ROOT / "custom_components" / "ha_creality_ws" / "sensor.py"
+    ).read_text(encoding="utf-8")
+    from custom_components.ha_creality_ws.utils import BUSY_PRINT_STATES
+
+    # derive_print_state is the only producer of these values.
+    utils_src = (
+        ROOT / "custom_components" / "ha_creality_ws" / "utils.py"
+    ).read_text(encoding="utf-8")
+    produced = set(re.findall(r'return "([a-z-]+)"', utils_src))
+    assert BUSY_PRINT_STATES <= produced, (
```

Ruff (0.16.1)

[warning] Yoda condition detected

Rewrite as `produced >= BUSY_PRINT_STATES`

(SIM300)